minoc/src/main/java/org/opencadc/minoc/ServiceAvailability.java

```diff
@@ -185,7 +196,27 @@ public AvailabilityStatus getStatus() {
      */
     @Override
     public void setState(String state) {
-        // ignore
+        String key = appName + RestAction.STATE_MODE_KEY;
+        if (RestAction.STATE_OFFLINE.equalsIgnoreCase(state)) {
+            System.setProperty(key, RestAction.STATE_OFFLINE);
+        } else if (RestAction.STATE_READ_ONLY.equalsIgnoreCase(state)) {
+            System.setProperty(key, RestAction.STATE_READ_ONLY);
+        } else if (RestAction.STATE_READ_WRITE.equalsIgnoreCase(state)) {
+            System.setProperty(key, RestAction.STATE_READ_WRITE);
+        } else {
+            throw new IllegalArgumentException("invalid state: " + state
+                + " expected: " + RestAction.STATE_READ_WRITE + "|" + RestAction.STATE_OFFLINE);
+        }
```

Comment:
STATE_READ_ONLY missing in the IllegalArgumentException expected message.

Reply:
Fixed. Good catch.